entity-framework/core/dbcontext-configuration/index.md

```diff
@@ -1,162 +1,393 @@
+The lifetime of a `DbContext` begins when the instance is created and ends when the instance is [disposed](https://docs.microsoft.com/dotnet/standard/garbage-collection/unmanaged). A `DbContext` instance is designed to be used for a _single_ [unit-of-work](https://www.martinfowler.com/eaaCatalog/unitOfWork.html). This means that the lifetime of a `DbContext` instance is usually very short.
 
-## Configuring DbContextOptions
+> [!TIP]
+> To quote Martin Fowler from the link above, "A Unit of Work keeps track of everything you do during a business transaction that can affect the database. When you're done, it figures out everything that needs to be done to alter the database as a result of your work."
 
-`DbContext` must have an instance of `DbContextOptions` in order to perform any work. The `DbContextOptions` instance carries configuration information such as:
+A typical unit-of-work when using Entity Framework Core (EF Core) involves:
 
-- The database provider to use, typically selected by invoking a method such as `UseSqlServer` or `UseSqlite`. These extension methods require the corresponding provider package, such as `Microsoft.EntityFrameworkCore.SqlServer` or `Microsoft.EntityFrameworkCore.Sqlite`. The methods are defined in the `Microsoft.EntityFrameworkCore` namespace.
-- Any necessary connection string or identifier of the database instance, typically passed as an argument to the provider selection method mentioned above
-- Any provider-level optional behavior selectors, typically also chained inside the call to the provider selection method
-- Any general EF Core behavior selectors, typically chained after or before the provider selector method
+- Creation of a `DbContext` instance
+- Tracking of entity instances by the context. Entities become tracked by
+  - Being [returned from a query](xref:core/saving/basic)
+  - Being [added or attached to the context](xref:core/saving/disconnected-entities)
+- Changes are made to the tracked entities as needed to implement the business rule
+- <xref:Microsoft.EntityFrameworkCore.DbContext.SaveChanges%2A> or <xref:Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync%2A> is called. EF Core detects the changes made and writes them to the database.
+- The `DbContext` instance is disposed
 
-The following example configures the `DbContextOptions` to use the SQL Server provider, a connection contained in the `connectionString` variable, a provider-level command timeout, and an EF Core behavior selector that makes all queries executed in the `DbContext` [no-tracking](xref:core/querying/tracking#no-tracking-queries) by default:
+> [!IMPORTANT]
+>
+> - It is very important to dispose the `DbContext` after use. This ensures both that any unmanaged resources are freed, and that any events or other hooks are unregistered so as to prevent memory leaks.
```

Comment:
```suggestion
> - It is very important to dispose the `DbContext` after use. This ensures both that any unmanaged resources are freed, and that any events or other hooks are unregistered so as to prevent memory leaks in case the instance remains referenced.
```